manifests/init.pp

```diff
@@ -39,61 +39,56 @@
 ) {
 
   # validate type and convert string to boolean if necessary
-  $package_latest_type = type($package_latest)
-  if $package_latest_type == 'string' {
+  if is_string($package_latest) == true {
     $my_package_latest = str2bool($package_latest)
   } else {
     $my_package_latest = $package_latest
   }
 
   # validate type and convert string to boolean if necessary
-  $service_running_type = type($service_running)
-  if $service_running_type == 'string' {
+  if is_string($service_running) == true {
     $my_service_running = str2bool($service_running)
   } else {
     $my_service_running = $service_running
   }
 
   # validate type and convert string to boolean if necessary
-  $service_hasstatus_type = type($service_hasstatus)
-  if $service_hasstatus_type == 'string' {
+  if is_string($service_hasstatus) == true {
     $my_service_hasstatus = str2bool($service_hasstatus)
   } else {
     $my_service_hasstatus = $service_hasstatus
   }
 
   # validate type and convert string to boolean if necessary
-  $service_hasrestart_type = type($service_hasrestart)
-  if $service_hasrestart_type == 'string' {
+  if is_string($service_hasrestart) == true {
     $my_service_hasrestart = str2bool($service_hasrestart)
   } else {
     $my_service_hasrestart = $service_hasrestart
   }
 
   if $peers != 'UNSET' {
-    $peers_type = type($peers)
-    case $peers_type {
-      'string': {
-        $my_peers = any2array($peers)
-        validate_array($my_peers)
-      }
-      'array': {
-        $my_peers = $peers
-        validate_array($my_peers)
-      }
-      'hash': {
-        $my_peers = $peers
-        validate_hash($my_peers)
-      }
-      default: {
-        fail("ntp::peers must be a string or an array or an hash. Detected type is <$peers_type>.")
-      }
+    if is_array($peers) == true {
+      $peers_type = 'array'
+      $my_peers = $peers
+      validate_array($my_peers)
+    }
+    elsif is_hash($peers) == true {
+      $peers_type = 'hash'
+      $my_peers = $peers
+      validate_hash($my_peers)
+    }
+    elsif is_string($peers) == true {
+      $peers_type = 'string'
+      $my_peers = any2array($peers)
+      validate_array($my_peers)
+    }
+    else {
+      fail("ntp::peers must be a string or an array or an hash.")
```

Comment:
we could add detecion for other (unsupported) types as well, but I don't think that's worth the effort here.